src/Visualization/Python/PlotMemoryMonitors.py

```diff
@@ -0,0 +1,269 @@
+        "The stylesheet can also be set with the 'SPECTRE_MPL_STYLESHEET' "
+        "environment variable."
+    ),
+)
+def plot_memory_monitors_command(
+    reduction_files: Sequence[str],
+    output: Optional[str],
+    use_mb: bool,
+    x_label: Optional[str],
+    x_bounds: Optional[Sequence[float]],
+    stylesheet,
+):
+    """
+    Plot the memory usage of a simulation from the MemoryMonitors data in the
+    Reductions H5 file.
+
+    This tool assumes there is a group in each of the "reduction-files" with the
+    path "/MemoryMonitors/" that holds dat files for each parallel component
+    that was monitored.
+
+    Note that the totals plotted here are not necessary the total memory usage
+    of the simulation. The memory monitors only capture what is inside
+    'pup'functions. Any memory that cannot be captured by a 'pup' function will
+    not by represented by this plot.
+    """
+
+    # Given an h5 file, make sure that the "MemoryMonitors" group exists. Then
+    # return the list of subfiles in that group
+    def check_memory_monitor_dir(h5_filename: str):
+        h5file = h5py.File(h5_filename, "r")
+        memory_monitor_dir = h5file.get("MemoryMonitors")
+        if memory_monitor_dir is None:
+            raise click.UsageError(
+                "Unable to open group 'MemoryMonitors' from h5 file"
+                f" {h5_filename}. Available subfiles are:\n"
+                f" {available_subfiles(h5file, extension='.dat')}"
+            )
+
+        return list(memory_monitor_dir.keys())
+
+    # Given an h5 file and a subfile, make sure that the subfile exists inside
+    # the h5 file. Then return the subfile as a DataFrame
+    def check_memory_monitor_file(
+        h5_filename: str, h5file: h5py.File, subfile_name: str
+    ):
+        subfile_path = f"/MemoryMonitors/{subfile_name}"
+        subfile = h5file.get(subfile_path)
+        if subfile_path is None:
+            raise click.UsageError(
+                f"Unable to open memory file '{subfile_path}'"
+                f" from h5 file {h5_filename}. Available subfiles are:\n"
+                f" {available_subfiles(h5file, extension='.dat')}"
+            )
+
+        return to_dataframe(subfile)
+
+    # Given a DataFrame, sum all the columns that list totals for each node
+    def total_over_nodes(df: pd.DataFrame):
+        total = None
+        for column in df.columns:
+            # "Size (MB)" is for a singleton. "Size on node" is for groups,
+            # nodegroups, and arrays
+            if column == "Size (MB)" or "Size on node" in column:
+                if total is not None:
+                    total += df[column]
+                else:
+                    # This copy is important because we want a new column/array.
+                    # Otherwise we'd just be storing a reference and we'd change
+                    # the values in 'df'.
+                    total = df[column].copy()
+
+        return total
```

Comment:
Want to sum over all columns? Then just do `df.sum(axis=1)`. Otherwise `df[cols_to_sum].sum(axis=1)`. Suggestion:

```py
def total_over_nodes(df):
    cols_to_sum = [col == "Size (MB)" or "Size on node" in col for col in df.columns]
    return df[cols_to_sum].sum(axis=1)
```